homework_18/Box.java

```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Box<T extends Fruit> {
+    private final List<T> fruitList = new ArrayList<>();
+
+    public void add(T fruit) {
+        Objects.requireNonNull(fruit);
+        fruitList.add(fruit);
```

Comment:
You should double-check whether the type of fruit is the same as in your current box.